plugins/inputs/smart/smart.go

```diff
@@ -62,6 +52,55 @@ var (
 		"194": "temp_c",
 		"199": "udma_crc_errors",
 	}
+
+	sasNvmeAttributes = map[string]struct {
+		ID    string
+		Name  string
+		Parse func(fields, deviceFields map[string]interface{}, str string) error
+	}{
+		"Accumulated start-stop cycles": {
+			ID:   "4",
+			Name: "Start_Stop_Count",
+		},
+		"Accumulated load-unload cycles": {
+			ID:   "193",
+			Name: "Load_Cycle_Count",
+		},
+		"Current Drive Temperature": {
+			ID:    "194",
+			Name:  "Temperature_Celsius",
+			Parse: parseTemperature,
+		},
+		"Temperature": {
+			ID:    "194",
+			Name:  "Temperature_Celsius",
+			Parse: parseTemperature,
+		},
+		"Power Cycles": {
+			ID:   "12",
+			Name: "Power_Cycle_Count",
+		},
+		"Power On Hours": {
+			ID:   "9",
+			Name: "Power_On_Hours",
+		},
+		"Media and Data Integrity Errors": {
+			Name: "Media_and_Data_Integrity_Errors",
+		},
+		"Error Information Log Entries": {
+			Name: "Error_Information_Log_Entries",
+		},
+		"Available Spare": {
+			Name: "Available_Spare",
+			Parse: func(fields, deviceFields map[string]interface{}, str string) error {
+				if str[len(str)-1] == '%' {
+					str = str[:len(str)-1]
```

Comment:
Bad things happen if str has length 0, I'd use strings.TrimSuffix here to make it a bit more readable.